File: map/map.dart
void main(List<String> args) {
  // Declaración
  final Map<String, dynamic> map = {
    "name": "Alvaro",
    "age": 25,
    "isStudent": true,
    "address": {"street": "Main Street", "city": "Lima", "country": "Peru"}
  };

  print(map);

  //Agregar
  map["lastName"] = "Huaysara";

  //Acceso
  print(map["name"]);
  print(map["address"]["city"]);

  //Claves
  print(map.keys);
  for (var key in map.keys) {
    print(key);
  }

  //Valores
  print(map.values);
  for (var value in map.values) {
    print(value);
  }

  //Items / Entries
  print(map.entries);
  for (var entry in map.entries) { //? no hay destructuring en dart
    // print(entry);
    print("Para la clave ${entry.key} se tiene el valor ${entry.value}");
  }
}

File: map/map.kt
fun main(args: Array<String>) {
    //Declaración (Inmutable)
    val map = mapOf(
        "name" to "Alvaro",
        "age" to 20,
        "address" to mapOf(
            "street" to "Calle 123",
            "city" to "Lima",
            "country" to "Perú"
        )
    );

    //Declaración (Mutable)
    val mutableMap: MutableMap<String, Any> = mutableMapOf(
        "name" to "Alvaro",
        "age" to 20,
        "address" to mutableMapOf<String, String>(
            "street" to "Calle 123",
            "city" to "Lima",
            "country" to "Perú"
        )
    );

    println(map);
    println(mutableMap);

    //Agregar
    mutableMap["lastName"] = "Huaysara"

    //Acceso
    println(mutableMap["name"])
    println((mutableMap["address"] as Map<*, *>)["city"])

    //Claves
    println(mutableMap.keys)
    for (key in mutableMap.keys) {
        println(key)
    }

    //Valores
    println(mutableMap.values)
    for (value in mutableMap.values) {
        println(value)
    }

    //Items / Entries
    println(mutableMap.entries)
    for ((key, value) in mutableMap.entries) {
        println("Para la clave $key se tiene el valor $value")
    }
}